shared/src/main/scala/coop/rchain/shared/PathOps.scala

```diff
@@ -38,4 +42,49 @@ object PathOps {
         FileVisitResult.CONTINUE
       }
     }
+
+  implicit class PathDelete(path: Path) {
+    def delete[F[_]: Sync: Log](): F[Unit] = {
+      import java.io.File
+      import java.util.Comparator
+
+      import scala.collection.JavaConverters._
+
+      import cats.implicits._
+
+      def deleteFile(file: File)(implicit logSource: LogSource): F[Unit] =
+        for {
+          deleted <- Sync[F].delay(file.delete)
+          _ <- if (deleted) Log[F].info(s"Deleted file ${file.getAbsolutePath}")
+              else Log[F].warn(s"Can't delete file ${file.getAbsolutePath}")
+        } yield ()
+
+      def getFiles: F[List[Path]] =
+        Sync[F].delay(
+          Files
+            .walk(path)
+            .sorted(Comparator.reverseOrder())
+            .iterator()
+            .asScala
+            .toList
+        )
+
+      def delete0(): F[Unit] =
+        for {
+          files <- getFiles
```

Comment:
what if `path` is a single file?

Reply:
this will be fine. I tested it in a side project